### DISCUSSION TITLE
Postal: Defining rail/puma

### DISCUSSION THREAD

**Discussion by drraccoony** [postalserver/postal#2585] [Help with using Postal]
We're running into the problem where rails is hitting "ActiveRecord Connection Pool Errors" but there appears to be no way out of the box in postal to define "RAILS_MAX_THREADS".

Am I missing a configuration option somewhere or is this not possible currently in postal?

> lib/postal/tracking_middleware.rb:34:in `dispatch_image_request'lib/postal/tracking_middleware.rb:21:in `call'104.28.58.4 - - [18/Aug/2023:16:23:06 +0000] "GET /500 HTTP/1.1" 500 1477 5.0037[10] [2023-08-18T16:23:06.843] FATAL -- :   ActiveRecord::ConnectionTimeoutError (could not obtain a connection from the pool within 5.000 seconds (waited 5.000 seconds); all pooled connections were in use):  lib/postal/tracking_middleware.rb:96:in `get_message_db_from_server_token'lib/postal/tracking_middleware.rb:75:in `dispatch_redirect_request'lib/postal/tracking_middleware.rb:25:in `call'35.150.70.104 - - [18/Aug/2023:16:23:06 +0000] "GET /500 HTTP/1.1" 500 1477 5.0046[10] [2023-08-18T16:23:06.847] FATAL -- :   ActiveRecord::ConnectionTimeoutError (could not obtain a connection from the pool within 5.000 seconds (waited 5.000 seconds); all pooled connections were in use):

**Comment by willpower232** [comment]
Have a look at https://github.com/postalserver/postal/pull/751#issuecomment-466540422 and https://github.com/postalserver/postal/issues/324#issuecomment-323538575
